controllers/btpoperator_controller.go

```diff
@@ -1192,10 +1298,85 @@ func (r *BtpOperatorReconciler) HandleReadyState(ctx context.Context, cr *v1alph
 	logger := log.FromContext(ctx)
 	logger.Info("Handling Ready state")
 
-	errorWithReason := r.chartConsistencyCheck(ctx, cr)
-	if errorWithReason != nil {
+	us, err := r.createUnstructuredObjectsFromManifestsDir(ctx, r.getResourcesToApplyPath())
+	if err != nil {
+		return r.UpdateBtpOperatorStatus(ctx, cr, types.StateError, CreatingObjectsFromManifestsFailed, fmt.Sprintf("Failed to create applicable objects from manifests: %s", err))
+	}
+	if errorWithReason := r.prepareModuleResources(ctx, us); errorWithReason != nil {
+		logger.Error(errorWithReason, "while preparing resources for provisioning")
 		return r.UpdateBtpOperatorStatus(ctx, cr, types.StateError, errorWithReason.reason, errorWithReason.message)
 	}
+	if err = r.reconcileResources(ctx, us); err != nil {
+		logger.Error(err, "while reconciling resources")
+		return r.UpdateBtpOperatorStatus(ctx, cr, types.StateError, ReconcileFailed, "Failed to reconcile resources")
+	}
+
+	return nil
+}
+
+func (r *BtpOperatorReconciler) addLabels(us ...*unstructured.Unstructured) {
+	for _, u := range us {
+		labels := u.GetLabels()
+		if len(labels) == 0 {
+			labels = make(map[string]string)
+		}
+		labels[managedByLabelKey] = operatorName
+		labels[chartVersionKey] = r.currentVersion
+		u.SetLabels(labels)
+	}
+}
+
+func (r *BtpOperatorReconciler) setNamespace(us ...*unstructured.Unstructured) {
+	for _, u := range us {
+		u.SetNamespace(ChartNamespace)
+	}
+}
 
+func (r *BtpOperatorReconciler) deleteCreationTimestamp(us ...*unstructured.Unstructured) {
+	for _, u := range us {
+		unstructured.RemoveNestedField(u.Object, "metadata", "creationTimestamp")
+	}
+}
+
+func (r *BtpOperatorReconciler) setConfigMapValues(secret *corev1.Secret, u *unstructured.Unstructured) error {
+	return unstructured.SetNestedField(u.Object, string(secret.Data["cluster_id"]), "data", "CLUSTER_ID")
+}
+
+func (r *BtpOperatorReconciler) setSecretValues(secret *corev1.Secret, u *unstructured.Unstructured) error {
+	for k := range secret.Data {
+		if err := unstructured.SetNestedField(u.Object, base64.StdEncoding.EncodeToString(secret.Data[k]), "data", k); err != nil {
+			return err
+		}
+	}
 	return nil
 }
+
+func (r *BtpOperatorReconciler) getResourcesToApplyPath() string {
+	return fmt.Sprintf("%s%capply", ResourcesPath, os.PathSeparator)
+}
+
+func (r *BtpOperatorReconciler) getResourcesToDeletePath() string {
```

Comment:
is this function used?
```
$ grep . -nrie 'getResourcesToDeletePath'
./controllers/btpoperator_controller.go:1358:func (r *BtpOperatorReconciler) getResourcesToDeletePath() string {
```

Reply:
not used yet, I will implement it in the next PR with changes to the update workflow